src/workerd/io/worker.c++

```diff
@@ -758,26 +759,22 @@ struct Worker::Script::Impl {
   // If set, then any attempt to use this script shall throw this exception.
   kj::Maybe<kj::Exception> permanentException;
 
-  void configureDynamicImports(jsg::Lock& js, jsg::ModuleRegistry& modules) {
-    struct DynamicImportResult {
-      jsg::Value value;
-      bool isException = false;
-    };
-
-    using Handler = kj::Function<jsg::Value()>;
-
-    static auto constexpr handleDynamicImport =
-        [](kj::Own<const Worker> worker,
-           Handler handler,
-           kj::Maybe<jsg::Ref<jsg::AsyncContextFrame>> asyncContext) ->
-               kj::Promise<DynamicImportResult> {
-      co_await kj::evalLater([] {});
-      auto asyncLock = co_await worker->takeAsyncLockWithoutRequest(nullptr);
-
-      Worker::Lock lock(*worker, asyncLock);
+  struct DynamicImportResult {
+    jsg::Value value;
+    bool isException = false;
+  };
+  using DynamicImportHandler = kj::Function<jsg::Value()>;
+
+  static DynamicImportResult getDynamicImportResult(
```

Comment:
I don't understand why this was refactored. Is it to avoid allocating `Worker::Lock` as a local variable in a coroutine? That was indeed a problem, but it looks like the change to use `jsg::runInV8Stack()` actually fixes the problem already -- the `Worker::Lock` is now in the inner callback, which is never a coroutine. So it seems like refactoring these functions wasn't really necessary? I worry that the refactoring makes the code a bit harder to read and adds some risk of breakage.

Reply:
I disagree. For myself, the slight refactoring on these makes it clearer that something is going on that I need to pay extra attention to, which I think is ideal. Unless there is something technically wrong with the refactor I prefer to keep these as is.